plugins/scaffolder-backend/src/scaffolder/stages/templater/rails.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ContainerRunner } from '@backstage/backend-common';
+import fs from 'fs-extra';
+import path from 'path';
+import { runCommand } from './helpers';
+import { TemplaterBase, TemplaterRunOptions } from './types';
+
+const commandExists = require('command-exists-promise');
+
+export class Rails implements TemplaterBase {
+  private readonly containerRunner: ContainerRunner;
+
+  constructor({ containerRunner }: { containerRunner: ContainerRunner }) {
+    this.containerRunner = containerRunner;
+  }
+
+  public async run({
+    workspacePath,
+    values,
+    logStream,
+  }: TemplaterRunOptions): Promise<void> {
+    const intermediateDir = path.join(workspacePath, 'intermediate');
+    await fs.ensureDir(intermediateDir);
+    const resultDir = path.join(workspacePath, 'result');
+
+    const { name, extraArguments, imageName, templateFile } = values;
+
+    // Directories to bind on container
+    const mountDirs = {
+      [workspacePath]: '/input',
+      [intermediateDir]: '/output',
+    };
+
+    const arrayExtraArguments = extraArguments?.split(',') || [];
+
+    const baseCommand = 'rails';
+    const baseArguments = ['new', '--skip-bundle', '--skip-webpack-install'];
+    const commandExistsToRun = await commandExists(baseCommand);
+
+    if (commandExistsToRun) {
+      if (templateFile) {
+        arrayExtraArguments.push('--template');
+        arrayExtraArguments.push(
+          templateFile.replace('./', `${workspacePath}/`),
+        );
+      }
+
+      await runCommand({
+        command: baseCommand,
+        args: [
+          ...baseArguments,
+          `${intermediateDir}/${name}`,
+          ...arrayExtraArguments,
+        ],
+        logStream,
+      });
+    } else {
+      if (templateFile) {
+        arrayExtraArguments.push('--template');
+        arrayExtraArguments.push(templateFile.replace('./', `/input/`));
+      }
+
+      await this.containerRunner.runContainer({
+        imageName: imageName || 'spotify/backstage-rails',
```

Comment:
The image is just a "suggestion", If you like to try you can use the dockerfile present here to build that.